src/interlocking_component/route_controller.py

```diff
@@ -33,55 +33,64 @@ class IInterlockingDisruptor:
     as well as to notify the Interlocking faults that occur in other parts of the simulation.
     """
 
-    def insert_track_blocked(self, track: "Track"):
+    def insert_track_blocked(self, track: Track):
         """This method is used to block a track and recalculate the routes of relevant trains.
 
-        :param track_id: the id of the blocked track
-        :type track_id: str
+        :param track: the blocked track
+        :type track: str
```

Comment:
Do you want the actual Track object or the id of that object? If the latter the `:type track:` has do be adjusted. (Applies also in the following methods)